### ISSUE TITLE
server/mssql: runtime errors on certain boolean conditions

### ISSUE THREAD

**Issue by 0x777** [hasura/graphql-engine#6944]
Create a table `author` as follows:
```sql
create table author ( id int not null primary key, name nvarchar not null);
```
Run this query:
```graphql
query {
  author(where: {_and:[] _or: []}) {
    name
  }
}
```
It fails with this error: 
```
An expression of non-boolean type specified in a context where a condition is expected, near 'AND'
```
The generated SQL is:
```sql
SELECT ISNULL((SELECT [t_author1].[name] AS [name]
FROM [dbo].[author] AS [t_author1]
WHERE (((0))
      AND ((1)))
FOR JSON PATH), '[]')
```
From [SQL Server's docs](https://docs.microsoft.com/en-us/sql/t-sql/language-elements/comparison-operators-transact-sql?view=sql-server-ver15): 
> The result of a comparison operator has the Boolean data type

I don't think there is any other way to create a value of 'Boolean' type. So maybe we should switch to `1=1` for `true` and `1<>1` for `false`?

**Comment by chrisdone** [comment]
I took a look; indeed. It's confusing, so for others reading later:

There is the BIT type, consisting of only 0 and 1. SQL Server will display 0 and 1 as True/False:

```
> select CAST('TRUE' as BIT);
[row 0]
: True
```

But try to use it in a boolean operator, and it is rejected:

```
> select 1 WHERE CAST('TRUE' as BIT) OR (1<2);
UnsuccessfulReturnCode "odbc_SQLExecDirectW" (-1) "[Microsoft][ODBC Driver 17 for SQL Server][SQL Server]An expression of non-boolean type specified in a context where a condition is expected, near 'OR'.
```

We have no proper boolean type. 

So the `(1=1)`/`(1<>1)` solution seems fine.

**Comment by 0x777** [comment]
Fixed in feb08d8998ef9f03470a0c54288d5155f2d40a05
